BrennanKolotinsky/hiive-auction-app
# [Feature] Devise + Item Model and Functionality

- setup Devise following this guide: [https://www.digitalocean.com/commu…](https://www.digitalocean.com/community/tutorials/how-to-set-up-user-authentication-with-devise-in-a-rails-7-application)
[…nity/tutorials/how-to-set-up-user-authentication-with-devise-in-a-rails-7-application](https://www.digitalocean.com/community/tutorials/how-to-set-up-user-authentication-with-devise-in-a-rails-7-application)
ideally would move the user sign in and sign out buttons to a React component, rather than  putting them in the layouts folder
- all routes point to home
- modify user factory to define necessary properties
- run command ```rails generate model Item name:string description:text user:references```
- add spec file to test new item model ensuring User is set
- adjust migration to support dropping table and requiring properties to be set
- adds the index route for displaying all Items
- creates controller and adds associated spec tests
- adds devise to rails_helper to allow access to current_user when running rspec
- adds a new post route for the Item controller, and adds associated testing
- adds a private method to only permit certain parameters
- add Item show route
- add show action in Item controller and set_item method called before this action as well as associated tests
- adds new Item and Items component displaying the contents of one item or multiple
- adds routes to these new components
- adds a services folder with two service get item which links to our show action and get items which links to our index action
- adds Item type
- adds new create item component that allows a user to input a name and description
- adds a route to reach this new component
- adds new service to call the api to create the item
- adds a button on the Items list to navigate to create an Item
- sets the Items list to the main component rendered on load (rather than guide code)
- adds a description of the project to the README.md
- adds a method to grab the latest auction and associated spec
- adds the ability to check whether a 30 second auction is currently running
- adds check to state when no items exist #self.is_active_auction? returns false
- adds private method no_active_auction that checks whether any auctions are active, and if called throws an error if there is an active auction
- adds validation to check #no_active_auction? is running and adds associated tests
- add home navigation
